GoTeamApp/GoTeamApp/SelectedLocationsManager.swift

```diff
@@ -31,6 +31,15 @@ class SelectedLocationsManager {
     }
   }
   
+  func update(location: Location) {
+    queue.async {
+      self.locations = self.locations.filter() { $0 == location }
```

Comment:
@wieniek, this will have the effect of reducing the self.locations to a single location, i.e. the passed in location. Since the 'location' is already a reference to the location there shouldn't be a need to apply a filter to self.locations to extract the location. I.e. you can directly do something like this:
func update(location: Location) {
    queue.async {
       self.dataStoreService.update(location: location)
   }
}
This should work as long as the update routine is called like this in the UI Layer.
var location = SelectedLocationManager.sharedInstance.locations[ix]
location.title = "New Title"
location.subtitle = "New address"
SelectedLocationManager.sharedInstance.update(location)

Let me know, I feel this should work.